netdata-installer.sh

```diff
@@ -413,6 +421,74 @@ fi
 
 trap build_error EXIT
 
+# -----------------------------------------------------------------------------
+
+fetch_libmosquitto() {
+  download_tarball "${1}" "${2}" "libmosquitto" "mosquitto"
+}
+
+build_libmosquitto() {
+  run make -C "${1}/lib"
+}
+
+copy_libmosquitto() {
+  target_dir="${PWD}/mosquitto"
+
+  run mkdir -p "${target_dir}"
+
+  run cp "${1}/lib/libmosquitto.a" "${target_dir}"
+  run cp "${1}/lib/*.h" "${target_dir}"
+}
+
+bundle_libmosquitto() {
+  if [ -n "${NETDATA_DISABLE_LIBMOSQUITTO}" ] ; then
+    return 0
+  fi
+
+  progress "Prepare custom libmosquitto version"
+
+  MOSQUITTO_PACKAGE_VERSION="$(cat packaging/mosquitto.version)"
+
+  tmp=$(mktemp -d /tmp/netdata-mosquitto-XXXXXX)
+  MOSQUITTO_PACKAGE_BASENAME="${MOSQUITTO_PACKAGE_VERSION}.tar.gz"
+
+  if [ -z "${NETDATA_LOCAL_TARBALL_OVERRIDE_MOSQUITTO}" ]; then
+    fetch_libmosquitto "https://github.com/netdata/mosquitto/archive/${MOSQUITTO_PACKAGE_BASENAME}" "${tmp}/${MOSQUITTO_PACKAGE_BASENAME}"
+  else
+    progress "Using provided mosquitto tarball ${NETDATA_LOCAL_TARBALL_OVERRIDE_MOSQUITTO}"
+    run cp "${NETDATA_LOCAL_TARBALL_OVERRIDE_MOSQUITTO}" "${tmp}/${MOSQUITTO_PACKAGE_BASENAME}"
+  fi
+
+  if [ ! -f "${tmp}/${MOSQUITTO_PACKAGE_BASENAME}" ] || [ ! -s "${tmp}/${MOSQUITTO_PACKAGE_BASENAME}" ]; then
+    run_failed "mosquitto download failed, MQTT ACLK will not be available"
```

Comment:
This will produce the wrong error message if the user supplied a path to override. Please consider changing it to a more general description or moving the error checking into both cases to make it more specific.

Reply:
Again, code copied from go.d.

Reply:
Also, why are we failing the install if the optional library could not be downloaded. We are supposed to continue without it.

Reply:
I misread the discussion in Slack.

Reply:
This actually isn't causing things to terminate. Relevant code is at https://github.com/netdata/netdata/blob/master/packaging/installer/functions.sh#L188-L190.

In short, the function being used is questionably named, but doesn't actually terminate execution, it just logs a failure message.